TraceListWidget.py

```diff
@@ -80,5 +87,49 @@ def popupMenu(self, pos):
                 fw = FitWindow(dataset, index, self)
                 self.windows.append(fw)
                 fw.show()
+
+            colorselctionAction = selectColorMenu.exec_(self.mapToGlobal(pos))
+
+            if colorselctionAction == redAction:
+                new_color = "r"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
+
+            if colorselctionAction == greenAction:
+                new_color = "g"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
+
+            if colorselctionAction == yellowAction:
+                new_color = "y"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
+
+            if colorselctionAction == cyanAction:
+                new_color = "c"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
+
+            if colorselctionAction == magentaAction:
+                new_color = "m"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
+
+            if colorselctionAction == whiteAction:
+                new_color = "w"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
                 
```

Comment:
For this code I would suggest having a single if statement and a dictionary example:

```python

coloractiondict = {magentaAction:'m', whitAction:'w'}

if colorselectionAction in coloractiondict.keys():
       new_color = coloractiondict[colorselectionAction]
....
...
...

```

Reply:
Okay. That will simplify the code. Will change it as requested.